reference.md

```diff
@@ -0,0 +1,1454 @@
+
+Try to register a snap without developer name or country: todo
+
+Try to register two snaps in a short period:
+
+    $ snapcraft register second-snap
+    Registering second-snap.
+    You must wait 152 seconds before trying to register your next snap.
+
+Try to register a reserved name:
+
+    $ snapcraft register bash
+    Registering bash.
+    The name 'bash' is reserved.
+
+    If you are the publisher most users expect for 'bash' then please claim the name at 'https://myapps.developer.ubuntu.com/dev/click-apps/register-name-dispute/?series=16&name=bash'
+
+Try to register a name already registered:
+
+    $ snapcraft register already-registered
+    Registering already-registered.
+    The name 'already-registered' is already taken.
+
+    We can if needed rename snaps to ensure they match the expectations of most users. If you are the publisher most users expect for 'already-registered' then claim the name at 'https://myapps.developer.ubuntu.com/dev/click-apps/register-name-dispute/?series=16&name=u1test201607183'
+
+Try to register a name with an invalid name:
+
+    $ snapcraft register _invalid   
+    Registering _invalid.
+    Registration failed.
+
+Try to push a file that doesn’t exist:
+
+    $ snapcraft push idontexist
+    The file 'idontexist' does not exist.
+
+Try to push a file that’s not a snap:
+
+    $ snapcraft push imnotasnap
+    Uploading imnotasnap.
+    Read on filesystem failed because EOF
+    Read on filesystem failed because EOF
+    Can't find a SQUASHFS superblock on imnotasnap
+    Command '['unsquashfs', '-d', '/tmp/tmphedels1u/squashfs-root', 'imnotasnap', '-e', 'meta/snap.yaml']' returned non-zero exit status 1
+
+Try to upload a snap not registered:
+
+    $ snapcraft push notregistered_version_arch.snap
+    Uploading notregistered_version_arch.snap.
+    Uploading notregistered_version_arch.snap [                                                 ]   0%
+    Uploading notregistered_version_arch.snap [=================================================] 100%
+    Sorry, try `snapcraft register notregistered` before pushing again.
+
+Upload a snap that requires a manual review:
+
+    $ snapcraft push manualreview_revision_architecture.snap
+    Uploading manualreview_revision_architecture.snap.
+    Uploading manualreview_revision_architecture.snap [                                      ]   0%
+    Uploading manualreview_revision_architecture.snap [======================================] 100%
+    Will need manual review...|
+    Revision 1 of 'manualreview_revision_architecture' created.
+    Publishing checks failed.
+    To release this to stable channel please request a review on the snapcraft list.
+    Use devmode in the edge or beta channels to disable confinement.
+
+Try to release a snap that doesn't exist:
+
+    $ snapcraft release idontexist 1 edge
+    Sorry, try `snapcraft register idontexist` before trying to release or choose an existing revision.
```

Comment:
I'm not sure why are we suggesting an existing revision. Shouldn't we suggest an existing snap instead?